CharlieBackend.Business/Services/ScheduleServiceFolder/EventsService.cs

```diff
@@ -58,36 +58,18 @@ public async Task<ScheduledEventDTO> GetAsync(long id)
             return _mapper.Map<ScheduledEventDTO>(await _unitOfWork.ScheduledEventRepository.GetByIdAsync(id));
         }
 
-        private async Task<string> ValidateUpdatedScheduleAsync(UpdateScheduledEventDto updatedSchedule)
-        {
-            if (await _unitOfWork.MentorRepository.GetByIdAsync(updatedSchedule.MentorId.GetValueOrDefault()) is null)
-            {
-                return ExceptionsConstants.MentorNotValid;
-            }
-            if (await _unitOfWork.ThemeRepository.GetByIdAsync(updatedSchedule.ThemeId.GetValueOrDefault()) is null)
-            {
-                return ExceptionsConstants.ThemeNotValid;
-            }
-            if (await _unitOfWork.StudentGroupRepository.GetByIdAsync(updatedSchedule.StudentGroupId.GetValueOrDefault()) is null)
-            {
-                return ExceptionsConstants.StudentGroupNotValid;
-            }
-
-            return string.Empty;
-        }
-
         public async Task<Result<ScheduledEventDTO>> ConnectScheduleToLessonById(long eventId, long lessonId)
         {
-            if (await _unitOfWork.ScheduledEventRepository.IsLessonConnectedToSheduledEventAsync(lessonId))
+            string errorMsg = await _validator.ValidateConnectEventToLessonAsync(eventId, lessonId);
+
+            if (!string.IsNullOrEmpty(errorMsg))
             {
-                return Result<ScheduledEventDTO>.GetError(ErrorCode.Conflict, $"Lesson with id={lessonId} is already associated with another ScheduledEvent");
+                throw new EntityValidationException(errorMsg);
             }
 
             var scheduleEntity = await _unitOfWork.ScheduledEventRepository.ConnectEventToLessonByIdAsync(eventId, lessonId);
 
-            return scheduleEntity == null ?
-                Result<ScheduledEventDTO>.GetError(ErrorCode.NotFound, $"Scheduled event with id={eventId} does not exist") :
-                Result<ScheduledEventDTO>.GetSuccess(_mapper.Map<ScheduledEventDTO>(scheduleEntity));
+            return Result<ScheduledEventDTO>.GetSuccess(_mapper.Map<ScheduledEventDTO>(scheduleEntity));
         }
 
         public async Task<Result<SingleEventDTO>> CreateSingleEventAsync(CreateSingleEventDto createSingleEvent)
```

Comment:
It may be better to use Result<ScheduledEventDTO> instead of throwing an exception, u sure that we need to throw an exception? It`s an additional performance issue, logging this exception and other effects